### ISSUE TITLE
S/N Error Array Discrepancy

### ISSUE THREAD

**Issue by sdahm** [Mosfire-DataReductionPipeline/MosfireDRP#4]
Reported by Joo Yoon (UCSB): The S/N calculated using the rms of an extracted spectrum is significantly different than the variance array produced by the MOSFIRE DRP. In this case, the S/N level of an emission line detection was 1.5 using the variance array and 12 when calculated directly from the spectrum. It should be noted that the observers anticipated a S/N detection level of 3-5 and believe the error array to be too high and the rms of the spectrum to be too low.

**Comment by MosfireDRP** [comment]
Thank you for reporting this issue. 
We will get back to you as soon as possible. It seems that a discussion of the way that the pipeline calculates the variance is in order.

**Comment by MosfireDRP** [comment]
Dear Scott and Joo Heon Yoon (@jhyoon79):

we have discussed this problem with the pipeline developers, and we have a few ideas to share with you:

First of all, it's important to understand exactly what is your definition of noise and how you measure it.
Our developer and collaborator Tuan Do (@followthesheep) comments: Can we investigate what methods and what is the definition that everyone is using? Are we talking about SNR per spectral channel? How is the noise determined in the final spectrum?
Chuck Steidel (@csteidel) also comments: I think we need to know more about what the data set looks like in this case, and why the variance array is being used rather than the sigma image (we went away from using the variance in the final reduced data back in April/May 2014, when a bug was corrected with the tracking of the noise in the final combined frame having to do with the fact that the data are reduced to e-/sec-- note that the files that should be looked at are the __eps.fits and _sig.fits, where the latter is not the variance, but is the 1 sigma noise (sqrt(sigma^2)). It is also relevant how many frames went into the dataset, as for small numbers of frames the current version of the drp combines the A and B frames in a non-optimal method. 
Incidentally, we (my group) have done a large amount of checking between the noise model and the actual noise in reduced data sets and have found very good agreement, hence the request for more information about the problem.
The noise model used for the DRP is quite simple, and is based on photon counting statistics (just the number of electrons per pixel) with the detector noise contribution added (assuming 21 e- single read, decreasing by a factor of N^0.5 where N is the number of reads being used) for each frame. The final *_sig.fits files are the noise level in e-/pixel, to go with the *_eps.fits files, which are the signal in e-/pixel. 
The meaning of the S/N in the snrs images is per pixel, so to get the S/N for a whole emission line one would have to account for the number of pixels,etc., but that should be easy enough with the drp output.


In summary, we are pretty confident that the pipeline is calculating the noise in a correct way, but we might just be using different definitions. Please take Chuck's consideration into account in comparing the results of your measurements with the pipeline's results.

Let us know if we can be of further assistance.

**Comment by jhyoon79** [comment]
Dear all,

I've found out that it's a simple documentation error.
The DRP manual (docx file) says filename_sig.fits is a variance but it should be sigma, standard deviation.
I am sorry for this hassle. The manual needs to be corrected.

Thanks
